### PR TITLE
[release_2.7] Remove get_platform_info API in ACRN DM

### PR DESCRIPTION
**Pull request by yonghuah**

### CODE REVIEW

**Comment by wenlingz** [comment]
@yakuizhao please help approve this PR, thanks

**Comment by wenlingz** [comment]
No new violations to the coding guideline detected.
No New Name Conflict
No New Function Declaration/Definition Mismatch

**Review by yakuizhao — APPROVED** [review]
LGTM on 2.7

**Comment by yonghuah** [comment]
"--cpu-affinity" is required after this PR#

File: devicemodel/hw/platform/acpi/acpi_parser.c [hunk 1]
```diff
@@ -0,0 +1,131 @@
+/*-
+ * Copyright (c) 2011 NetApp, Inc.
```
Comment (gvancuts) [inline devicemodel/hw/platform/acpi/acpi_parser.c:2]: We seem to have a number of ACRN-specific modifications that were made to this file. Shouldn't the Intel Copyright notice have been added?

File: devicemodel/hw/platform/acpi/acpi_parser.c [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	}
+
+	if (pcpu_num == 0) {
+		ret = -1;
+	}
+	return ret;
+}
+
+/*
+ * There has an assumption. The SOS owned pcpu starts from physical cpu 0,
```
Comment (gvancuts) [inline devicemodel/hw/platform/acpi/acpi_parser.c:96]: Update to: `There is an assumption: the Service VM owns the Physical CPU (cCPU) starting from physical CPU #0.

File: devicemodel/hw/platform/acpi/acpi_parser.c [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+ * otherwise SOS doesn't know the mapping relationship between its vcpu0
```
Comment (gvancuts) [inline devicemodel/hw/platform/acpi/acpi_parser.c:97]: Update `SOS` to `Service VM`

File: devicemodel/hw/platform/acpi/acpi_parser.c [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+int parse_madt(void)
+{
+	int ret = 0U;
+	ssize_t size;
+	struct acpi_table_madt *madt;
+	struct stat file_state;
+
+	int fd = open("/sys/firmware/acpi/tables/APIC", O_RDONLY);
+	if (fd < 0) {
+		pr_err("Fail to open sos APIC file!\n");
```
Comment (gvancuts) [inline devicemodel/hw/platform/acpi/acpi_parser.c:109]: Update error message to: `Could not open the Service VM APIC file: /sys/firmware/acpi/tables/APIC`

File: devicemodel/hw/platform/acpi/acpi_parser.c [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+		return -1;
+	}
+
+	if (fstat(fd, &file_state) == -1) {
+		pr_err("Fail to get apic file state!\n");
```
Comment (gvancuts) [inline devicemodel/hw/platform/acpi/acpi_parser.c:114]: Update to; `Could not retrieve the APIC file state (/sys/firmware/acpi/tables/APIC)`

File: devicemodel/hw/platform/acpi/acpi_parser.c [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+		close(fd);
+		return -1;
+	}
+
+	madt = (struct acpi_table_madt *)malloc(file_state.st_size);
+	size = read(fd, madt, file_state.st_size);
+	if (size == file_state.st_size) {
+		ret = local_parse_madt(madt);
+	} else {
+		pr_err("Fail to read sos madt info!");
```
Comment (gvancuts) [inline devicemodel/hw/platform/acpi/acpi_parser.c:124]: Update to: `Could not read the Service VM MADT information`

File: devicemodel/hw/platform/acpi/rtct.c [hunk 7]
```diff
@@ -409,7 +444,11 @@ static int init_guest_lapicid_tbl(struct acrn_platform_info *platform_info, uint
 		if (pcpu_id < 0)
 			return -1;
 
-		guest_lapicid_tbl[vcpu_id] = lapicid_from_pcpuid(platform_info, pcpu_id);
+		guest_lapicid_tbl[vcpu_id] = lapicid_from_pcpuid(pcpu_id);
+		if (guest_lapicid_tbl[vcpu_id] == -1) {
+			pr_err("Get lapic id fail.\n");
```
Comment (gvancuts) [inline devicemodel/hw/platform/acpi/rtct.c:449]: Update to: `Could not retrieve local APIC ID.`

File: devicemodel/core/main.c [hunk 8]
```diff
@@ -1012,6 +1012,11 @@ main(int argc, char *argv[])
 		exit(1);
 	}
 
+	if (parse_madt()) {
+		pr_err("parse madt failed\n");
```
Comment (gvancuts) [inline devicemodel/core/main.c:1016]: Update to: `Failed to parse MADT`

File: devicemodel/hw/platform/acpi/acpi.c [hunk 9]
```diff
@@ -343,7 +323,11 @@ basl_fwrite_madt(FILE *fp, struct vmctx *ctx)
 			return -1;
 		}
 
-		lapic_id = lapicid_from_pcpuid(&plat_info, pcpu_id);
+		lapic_id = lapicid_from_pcpuid(pcpu_id);
+		if (lapic_id == -1) {
+			pr_err("Get lapic id fail.\n");
```
Comment (gvancuts) [inline devicemodel/hw/platform/acpi/acpi.c:328]: Update to: `Could not retrieve local APIC ID.`

**Pull request merged**